### PR TITLE
Generate and display blockies for wallets #2294

### LINKED ISSUE #2294: Generate and display blockies for wallets
Blockies like https://github.com/ethereum/blockies. Maybe there's a Swift version?

2 places:

* Left of navigation bar in Wallet tab (screenshot below)
* Rows in "Change Wallet" screen zpl.io/bWn60RP

image

### PR DESCRIPTION
**Pull request by vladyslav-iosdev**
Closes #2294

### CODE REVIEW

**Comment by James-Sangalli** [comment]
@vladyslav-iosdev missing in the change wallet page

File: AlphaWallet/Core/Helpers/BlockieGenerator.swift [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+import Foundation
+import BlockiesSwift
+import PromiseKit
+import UIKit.UIImage
+
+typealias BlockiesImage = UIImage
+
+class BlockiesGenerator {
+    private static var cache: [String: BlockiesImage] = [:]
```
Comment (hboon) [inline AlphaWallet/Core/Helpers/BlockieGenerator.swift:16]: The key can be `AlphaWallet.Address`.

File: AlphaWallet/Core/Helpers/BlockieGenerator.swift [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    private func createBlockiesImage(address: AlphaWallet.Address) -> Promise<BlockiesImage> {
+        enum AnyError: Error {
+            case blockieCreateFailure
+        }
+
+        return Promise { seal in
+            let blockies = Blockies(seed: address.eip55String, size: 8, scale: 3)
+            if let image = blockies.createImage() {
```
Comment (hboon) [inline AlphaWallet/Core/Helpers/BlockieGenerator.swift:65]: This takes about ~2s when I run it so it blocks while the app is launching. Can we make it run in the background?

**Review by hboon — CHANGES_REQUESTED** [review]

**Review by hboon — CHANGES_REQUESTED** [review]
> missing in the change wallet page

@vladyslav-iosdev don't forget that one

**Review by hboon — APPROVED** [review]

**Pull request merged**